docs/core/tools/create-custom-templates.md

```diff
@@ -0,0 +1,232 @@
+---
+title: Create custom templates for dotnet new | Microsoft Docs
+description: This topic shows you how to create custom templates for any type of .NET project or files.
+keywords: dotnet new, CLI, CLI command, .NET Core, template, templating
+author: guardrex
+ms.author: mairaw
+ms.date: 06/26/2017
+ms.topic: article
+ms.prod: .net-core
+ms.technology: dotnet-cli
+ms.devlang: dotnet
+ms.assetid: c28eb59d-4d18-4aba-bcd1-88bd887abc08
+---
+
+# Create custom templates for dotnet new
+
+[!INCLUDE [core-preview-warning](~/includes/core-preview-warning.md)]
+
+The [.NET Core SDK](https://www.microsoft.com/net/download/core) comes with many templates pre-installed for you to use with the [`dotnet new` command](dotnet-new.md), but you can create your own custom templates for any type of .NET project, such as an app, service, tool, or class library. You can even create a template that outputs one or more independent files, such as a configuration file.
+
+Users install your custom template from a NuGet package on any NuGet feed, by referencing a NuGet *nupkg* file directly, or by specifying a file system directory that contains the template. The template engine offers features that allow you to replace values, include and exclude files and regions of files, and execute custom processing operations when your template is used.
+
+The template engine is open source, and the online code repository is at [dotnet/templating](https://github.com/dotnet/templating/) on GitHub. Visit the [dotnet/dotnet-template-samples](https://github.com/dotnet/dotnet-template-samples) repo for samples of templates. More templates, including templates from third parties, are found at [Available templates for dotnet new](https://github.com/dotnet/templating/wiki/Available-templates-for-dotnet-new) on GitHub. A detailed pre-release blog post that describes how to create and install custom templates is at [How to create your own templates for dotnet new](https://blogs.msdn.microsoft.com/dotnet/2017/04/02/how-to-create-your-own-templates-for-dotnet-new/).
```

Comment:
Yes, you're correct: We decided in a short offline chat (I'll add a note to the issue) that we need to keep the blog post linked here for two reasons:

* For the first pass on fully covering templates, we're covering the utter basics of template creation and usage and not adding a tutorial/walk-through to go with this reference topic. I did one thing that helps a dev trying to create a basic template: I kept the example "Garcia Software Console" template in use throughout the topic. This topic is really only going to get us started. We need a real tutorial, and I'll open an issue after we get this merged to discuss this aspect further.
* The first version of this topic won't cover all of the features of template design, so we'll need to expand the coverage. Where and how to cover the finer details for features such as replaceable parameter values, optional content, parameter options, custom processing, etc. is TBD. When I open the issue later, let's discuss how to best layout the coverage.

We'll probably end up with two issues: one saying we need a tutorial and a second saying we need to cover additional features of template design and how to lay that topic (or topics) out.

What I'll do here is just instruct the reader to access the blog post and Wiki for additional info after they've read this topic. (Let's take a 1st look at some draft language on the next commit, and then we can modify as needed.) As the reader gets down into the topic, they have linked access to  the *template.json* schema. When the features of templates are fully in place in the docs, then yes, I think we should drop the blog post entirely.